### ISSUE TITLE
Extremely slow lora training on colab A100 instance (wan video)

### ISSUE THREAD

**Issue by Homagn** [modelscope/DiffSynth-Studio#435]
Currently testing this finetuning script

# train
!CUDA_VISIBLE_DEVICES="0" python train_wan_t2v.py \
  --task train \
  --train_architecture lora \
  --dataset_path custom_dataset/mydataset \
  --output_path ./models \
  --dit_path "models/Wan-AI/Wan2.1-T2V-1.3B/diffusion_pytorch_model.safetensors" \
  --steps_per_epoch 500 \
  --max_epochs 10 \
  --learning_rate 1e-4 \
  --lora_rank 16 \
  --lora_alpha 16 \
  --lora_target_modules "q,k,v,o,ffn.0,ffn.2" \
  --accumulate_grad_batches 1 \
  --use_gradient_checkpointing

Everything seems to be working fine, except the logs kind of scare me (log below)
Based on displayed 8.30s/it it will take 1 hour to train 1 epoch, which means 10 hours to train 10 epochs? correct me if I'm wrong, but is there any way to speed it up? (logs below)

2025-03-11 04:00:27.915618: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-03-11 04:00:27.933281: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1741665627.954840   11531 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1741665627.961502   11531 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-03-11 04:00:27.983232: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
7 videos in metadata.
7 tensors cached in metadata.
Loading models from: models/Wan-AI/Wan2.1-T2V-1.3B/diffusion_pytorch_model.safetensors
    model_name: wan_video_dit model_class: WanModel
        This model is initialized with extra kwargs: {'has_image_input': False, 'patch_size': [1, 2, 2], 'in_dim': 16, 'dim': 1536, 'ffn_dim': 8960, 'freq_dim': 256, 'text_dim': 4096, 'out_dim': 16, 'num_heads': 12, 'num_layers': 30, 'eps': 1e-06}
    The following models are loaded: ['wan_video_dit'].
No wan_video_text_encoder models available.
Using wan_video_dit from models/Wan-AI/Wan2.1-T2V-1.3B/diffusion_pytorch_model.safetensors.
No wan_video_vae models available.
No wan_video_image_encoder models available.
/usr/local/lib/python3.11/dist-packages/lightning/fabric/connector.py:572: `precision=bf16` is supported for historical reasons but its usage is discouraged. Please set your precision to bf16-mixed instead!
Using bfloat16 Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
You are using a CUDA device ('NVIDIA A100-SXM4-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name | Type             | Params | Mode
-------------------------------------------------
0 | pipe | WanVideoPipeline | 1.4 B  | eval
-------------------------------------------------
21.9 M    Trainable params
1.4 B     Non-trainable params
1.4 B     Total params
5,763.478 Total estimated model params size (MB)
3677      Modules in train mode
1         Modules in eval mode
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/trainer/connectors/data_connector.py:425: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
Epoch 0:  16% 80/500 [11:03<58:05,  8.30s/it, v_num=1, train_loss=0.0993]^C

**Comment by Artiprocher** [comment]
@Homagn I believe this speed is normal. This model consumes a significant amount of GPU memory due to the large activation values during video training. If you want to pursue lightweight training, we recommend reducing the number of frames or directly using image training.

**Comment by Homagn** [comment]
Thanks for the clarification, for the image training do I just need to add this argument as mentioned in the readme ?

- "If you wish to train the image-to-video model, please add an extra parameter --image_encoder_path "models/Wan-AI/Wan2.1-I2V-14B-480P/models_clip_open-clip-xlm-roberta-large-vit-huge-14.pth"."

Also before trying just wanted to make sure, would it be possible to fit in memory with the 40GB vram for A100 on colab?

**Comment by Artiprocher** [comment]
@Homagn For the first question, yes. Our code will automatically detect the model type when you provide the i2v model path. For the second question, I think 40GB is not enough for training. 80GB is recommended.

**Comment by tomasctg** [comment]
@Artiprocher  Is it possible to train the 14B I2V model with LoRA quantized in FP8? If it is possible, what would be the bash file to do that?
